### ISSUE TITLE
Some General Feedback on Deck for Android

### ISSUE THREAD

**Issue by aljawaid** [stefan-niedermann/nextcloud-deck#1290]
Hello, fantastic app, excellent for fast-moving workflow, but below are my suggestions and challenges, if anybody would like some feedback:

- Mobile notifications - excellent, but...
- - self-made comments shouldnt be a notification to oneself. or mute own comment notifications if deck server-side issue. I think when a comment is added to the card, it sends notifications to all users, even the comment creator
- - clicking on notification should go direct to the edit.... if comment is made, then go direct to comment... not the card

- Add voice note capability
-- my very small team, some of them have voice notes and audio recordings forwarded through whatsapp... would be good to play those forwarded files within the app inline
-- a better emoji icon for comments
-- add voice note direct in comments

- Clone Cards
-- cards can be moved but not cloned currently... let them be cloned, just like the app allows boards to be cloned

- Default Board
-- Allow option to select a default board, or the upcoming cards on startup

- Delete Attachments (mentioned elsewhere too by others)
-- I dont know if this is a bug, but sometimes it double-uploads the picture we share from whatsapp to a card as it displays an error message `failed to delete local file`... so when we open the app again, i think it uploads a second copy... problem is, it cant be deleted without going directly to the file through teh nextcloud files app.
-- Allow to rename attachments

- Tags
-- Allow tags to be set at global level also.... or display a screen of all the existings tags for which boards in a table so we can manually copy them and duplicate them to similar boards

- Share
-- The share content is really good, we get to share the title/description/tags/date created in text format through whatsapp etc
-- Allow to share/save/export comments as pdf
-- Allow to save/export full card+comments+list of attachments as pdf

- Archived stuff
-- Have a filter to show active/archived cards on the same screen maybe?

- Comments
-- Allow entering a new line in comments using the enter button
-- Allow admins to delete comments
-- Just like inline reply to comment, allow inline forward comment so we can tag a user+message related to a reply.... a sort of soft-assign one chat to another within the comment. so John makes a comment, Supervisor Jack inline forwards comment to Susan for follow up. The actual card is still assigned to John.  Extended option would be to allow automatically adding Susan to assignment if inline forward tagged.

Regardless of the above, we are very much looking forward to updates to the app as the mobile client is crucial to make Deck productive so thanks a lot for your hard work.

**Comment by stefan-niedermann** [comment]
Whoa, that's some big feedback - thank you very much for that!

Please give me some time to analyze the issue and write you a peoper response later today 🙂

**Comment by stefan-niedermann** [comment]
First of all: We are developing the Deck Android app as an independent team - we do not work for or are related in any way to the Nextcloud GmbH (which develops the Deck server app).

> Mobile notifications - excellent, but...
>  self-made comments shouldnt be a notification to oneself. or mute own comment notifications if deck server-side issue. I think when a comment is added to the card, it sends notifications to all users, even the comment creator

➡️ Please open an issue in the [repository of the Deck server app](https://github.com/nextcloud/deck/issues/new/choose) which generates the notifications

>    clicking on notification should go direct to the edit.... if comment is made, then go direct to comment... not the card

➡️ Reasonable Please split to a separate issue in our repository

> Add voice note capability
> -- my very small team, some of them have voice notes and audio recordings forwarded through whatsapp... would be good to play those forwarded files within the app inline

➡️ We have plans to allow users to make voice recordings (and a dedicated picker) in the `Attachments` section. Since we don't have an issue for this yet, feel free to open one for transparency reasons.

> -- a better emoji icon for comments

➡️ What's wrong with the current icon for comments? What do you suggest especially? I am curious what icon could better represent the term `Comment` than the actual Google Material Icon for `Comment`?

![image](https://user-images.githubusercontent.com/4741199/173758130-7c3bc7e6-5782-4a7e-9517-5c0d6700f7a3.png)

> -- add voice note direct in comments

➡️ `Comments` currently are limited to text only (and a little bit markdown, which boils down to plain text after all) at the moment by the server side. I am not sure, but I guess, that you will need to open an issue in the [Nextcloud repository](https://github.com/nextcloud/server/issues/new/choose) itself because I believe that the Deck server app reuses this part.

> Clone Cards
> -- cards can be moved but not cloned currently... let them be cloned, just like the app allows boards to be cloned

➡️ Reasonable Please split to a separate issue in our repository. Beware though, that this will probably *not* allow cloning attachments or comments just like the clone feature when cloning a board together with its cards.

> Default Board
> -- Allow option to select a default board, or the upcoming cards on startup

➡️ Reasonable Please split to a separate issue in our repository.

> Delete Attachments (mentioned elsewhere too by others)

➡️ We supported this feature but [Deck server 1.3.x introduced a bug](https://github.com/nextcloud/deck/issues/2904) on the server side (and is still not fixed until today) which makes it impossible for clients to delete attachments. We therefore disable this feature in the Deck Android app until the actual cause has been fixed. Feel free to upvote / comment the linked issue :shrug: There's nothing we can do on our side.

> -- I dont know if this is a bug, but sometimes it double-uploads the picture we share from whatsapp to a card as it displays an error message failed to delete local file... so when we open the app again, i think it uploads a second copy... problem is, it cant be deleted without going directly to the file through teh nextcloud files app.

➡️ You already [found the issue in our issue tracker as it looks](https://github.com/stefan-niedermann/nextcloud-deck/issues/1221#issuecomment-1156011906)

> -- Allow to rename attachments

➡️ Please open an issue in the [repository of the Deck server app](https://github.com/nextcloud/deck/issues/new/choose) which currently does not provide a public REST-API endpoint to rename attachments

> Tags
> -- Allow tags to be set at global level also.... or display a screen of all the existings tags for which boards in a table so we can manually copy them and duplicate them to similar boards

➡️ This is [blocked by the Deck server app](https://github.com/nextcloud/deck/issues/994) and nothing we can change on our side.

> Share
> -- The share content is really good, we get to share the title/description/tags/date created in text format through whatsapp etc
> -- Allow to share/save/export comments as pdf
> -- Allow to save/export full card+comments+list of attachments as pdf

➡️ [We do not have any plans to implement PDF exports](https://github.com/stefan-niedermann/nextcloud-deck/issues/1192#issuecomment-1012064859) for the Deck Android app.

> Archived stuff
> -- Have a filter to show active/archived cards on the same screen maybe?

➡️ [Blocked by the Deck Server app](https://github.com/nextcloud/deck/issues/2613)

> Comments
> -- Allow entering a new line in comments using the enter button

➡️ This is technically possible and supported by the Deck server app using <kbd>Shift</kbd> + <kbd>Enter</kbd>. Please split to a separate issue in our repository

> -- Allow admins to delete comments

➡️ @desperateCoder please check whether this is possible API wise. If not, we will need to check where this issue can be adressed. I do not have any strong feelings to block this, but my personal opinion is: Usually you have an organizational issue in your team if an administrator has to delete team member comments like a moderator in an forum :smile: 

> -- Just like inline reply to comment, allow inline forward comment so we can tag a user+message related to a reply.... a sort of soft-assign one chat to another within the comment. so John makes a comment, Supervisor Jack inline forwards comment to Susan for follow up. The actual card is still assigned to John. Extended option would be to allow automatically adding Susan to assignment if inline forward tagged.

➡️ First of all: Currently only one assignee is able per card. Everything else requires changes to the Deck server app. Looks like you are mixing up comment forwarding with assignees with using comments as some kind of chat system to cover your workflow. Honestly, that sounds a bit strange to me, maybe I just don't get the idea at all. But I'll pull the joker here: The Deck Android app will follow the Deck server app in mostly every use case and tries to be consistent. Address your use case / workflow there and if it will be implemented, we will definitely follow :wink:

**Comment by aljawaid** [comment]
Hello, sorry for the delay in responding... I managed to go through your response and I have linked what and where you advised.  I look forward to any of my issues or suggestions being taken further.

> What's wrong with the current icon for comments? What do you suggest especially? I am curious what icon could better represent the term Comment than the actual Google Material Icon for Comment?
- _I meant the emojis from the keyboard for the comments section. I was expecting common emojis to be shown but it showed the default microsoft keyboard ones which were ugly._

> but my personal opinion is: Usually you have an organizational issue in your team if an administrator has to delete team member comments like a moderator in an forum
- _exactly_

> First of all: Currently only one assignee is able per card. Everything else requires changes to the Deck server app. Looks like you are mixing up comment forwarding with assignees with using comments as some kind of chat system to cover your workflow. Honestly, that sounds a bit strange to me, maybe I just don't get the idea at all. But I'll pull the joker here: The Deck Android app will follow the Deck server app in mostly every use case and tries to be consistent. Address your use case / workflow there and if it will be implemented, we will definitely follow
- _Just to explain, what I meant was that we can reference/reply to a comment which shows in the flow of the comments... but sometimes in a comment we want one reply to be 'forwarded/mentioned/referenced' along to another user for action. the task can have more than one assignee in deck mobile, but even with that... now just THIS replied comment, I want user 2 only to action (in return user 2 will get a mobile notification as such)... while user 1 and 2 overall manage the card.  Maybe its just my use case but I thought I'd explain further. It's not a critical feature._

You can close this issue if you think appropriate as I have linked all the doable stuff.  Thanks for listening though.

**Comment by stefan-niedermann** [comment]
Thanks for splitting this up into separate issues :slightly_smiling_face: Closing this one in favour of the follow-up issues.
